File: 1695-maximum-erasure-value/1695-maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        hset = set()
        i = 0
        currsum = 0
        result = 0
        for j in range(len(nums)):
            # handle when duplicates exist
            while nums[j] in hset:
                hset.remove(nums[i])
                currsum -= nums[i]
                i+=1
            hset.add(nums[j])
            currsum += nums[j]
            result = max(currsum,result)
        return result

File: 1695-maximum-erasure-value/1695-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        unordered_set<int> hset;
        int result = 0,sum = 0;
        int i=0;
        for(int j=0;j<nums.size();j++) {
            //check elem in set and update
            while(hset.find(nums[j]) != hset.end()) {
                hset.erase(nums[i]);
                sum -= nums[i];
                i++;
            }
            hset.insert(nums[j]);
            sum += nums[j];
            result = max(sum,result);
        }
        return result;
    }
};